### PR TITLE
Gracefully handle unknowns in filters in backend

### PR DESCRIPTION
**Pull request by pfeeny5**
## Issue

- Addresses #232 
## Description

Allow filters to include unknown entires for incomplete data in the backend.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Visit localhost:3100 for backend.  Test following URLs to see data.
Toggle between True and False at the end:
http://localhost:3100/listings?filter[0][$comparison]==&filter[0][minRent]=5&filter[0][$include_nulls]=false
http://localhost:3100/listings?filter[0][$comparison]==&filter[0][minRent]=5&filter[0][$include_nulls]=true
http://localhost:3100/listings?filter[0][$comparison]=NA&filter[0][ami]=60&filter[0][$include_nulls]=false
http://localhost:3100/listings?filter[0][$comparison]=NA&filter[0][ami]=60&filter[0][$include_nulls]=true

Additional testing has been added to listing.service.spec.ts.

- [x] Desktop View
- [ ] Mobile View
- [ ] Test A
- [ ] Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes
- [ ] I have run `yarn generate:client` if I made backend changes

### CODE REVIEW

File: backend/core/src/listings/listings.service.spec.ts [hunk 1]
```diff
@@ -194,6 +196,56 @@ describe("ListingsService", () => {
       )
     })
 
+    it("should support basic filters with null data", async () => {
```
Comment (avaleske) [inline backend/core/src/listings/listings.service.spec.ts:199]: nit: how about "should include listings with missing data if $include_nulls is true"

File: backend/core/src/listings/listings.service.spec.ts [hunk 2]
```diff
@@ -194,6 +196,56 @@ describe("ListingsService", () => {
...
+      expect(listings.items).toEqual(mockListings)
+      expect(mockInnerQueryBuilder.andWhere).toHaveBeenCalledWith(
+        "(LOWER(CAST(listings.name as text)) = LOWER(:name_0) OR listings.name IS NULL)",
+        {
+          name_0: "minRent",
+        }
+      )
+    })
+
+    it("should support custom filters with null data", async () => {
```
Comment (avaleske) [inline backend/core/src/listings/listings.service.spec.ts:224]: similar nit here: can we name this slightly more descriptively so it answers: What does "support" mean in this context?

File: backend/core/src/shared/filter/custom_filters.ts [hunk 3]
```diff
@@ -5,16 +5,25 @@ import {
...
   const reservedCommunityTypeColumnName = `LOWER(CAST(${
     filterTypeToFieldMap[ListingFilterKeys.seniorHousing]
   } as text))`
   if (filterValue == "true") {
-    qb.andWhere(`${reservedCommunityTypeColumnName} = LOWER(:${whereParameterName})`, {
-      [whereParameterName]: seniorHousingCommunityType,
-    })
+    qb.andWhere(
+      `(${reservedCommunityTypeColumnName} = LOWER(:${whereParameterName})  ${
+        includeNulls ? `OR ${reservedCommunityTypeColumnName} IS NULL` : ""
```
Comment (avaleske) [inline backend/core/src/shared/filter/custom_filters.ts:21]: very nit: Here and elsewhere, can we move the extra space inside the extra string, right before the OR

**Review by avaleske — APPROVED** [review]
Thank you! LGTM but had a few naming nits.

**Review by willrlin — APPROVED** [review]
assuming you've addressed Austin's comments

File: backend/core/src/listings/listings.service.spec.ts (hunk 1 above)
Comment (pfeeny5) [reply backend/core/src/listings/listings.service.spec.ts:199]: updated

**Pull request merged**